docs/dapps/interact-with-contracts.md

```diff
@@ -0,0 +1,266 @@
+
+```rust
+import "std/fungible_token_interface"
+
+// Defines a contract named `TokenFaucet`.
+// A contract is a collection of fields (its state) and functions.
+// Once deployed, a contract resides at a specific address on the Alephium blockchain.
+// Contract fields are permanently stored in contract storage.
+// A contract can issue an initial amount of token at its deployment.
+Contract TokenFaucet(
+    symbol: ByteVec,
+    name: ByteVec,
+    decimals: U256,
+    supply: U256,
+    mut balance: U256
+) implements IFungibleToken {
+
+    // Events allow for logging of activities on the blockchain.
+    // Alephium clients can listen to events in order to react to contract state changes.
+    event Withdraw(to: Address, amount: U256)
+
+    enum ErrorCodes {
+        InvalidWithdrawAmount = 0
+    }
+
+    // A public function that returns the initial supply of the contract's token.
+    // Note that the field must be initialized as the amount of the issued token.
+    pub fn getTotalSupply() -> U256 {
+        return supply
+    }
+
+    // A public function that returns the symbol of the token.
+    pub fn getSymbol() -> ByteVec {
+        return symbol
+    }
+
+    // A public function that returns the name of the token.
+    pub fn getName() -> ByteVec {
+        return name
+    }
+
+    // A public function that returns the decimals of the token.
+    pub fn getDecimals() -> U256 {
+        return decimals
+    }
+
+    // A public function that returns the current balance of the contract.
+    pub fn balance() -> U256 {
+        return balance
+    }
+
+    // A public function that transfers tokens to anyone who calls it.
+    // The function is annotated with `updateFields = true` as it changes the contract fields.
+    // The function is annotated as using contract assets as it does.
+    @using(assetsInContract = true, updateFields = true, checkExternalCaller = false)
+    pub fn withdraw(amount: U256) -> () {
+        // Debug events can be helpful for error analysis
+        emit Debug(`The current balance is ${balance}`)
+
+        // Make sure the amount is valid
+        assert!(amount <= 2, ErrorCodes.InvalidWithdrawAmount)
+        // Functions postfixed with `!` are built-in functions.
+        transferTokenFromSelf!(callerAddress!(), selfTokenId!(), amount)
+        // Ralph does not allow underflow.
+        balance = balance - amount
+
+        // Emit the event defined earlier.
+        emit Withdraw(callerAddress!(), amount)
+    }
+}
+```
+</p></details>
+
+The `TokenFaucet` contract has `5` public functions that only reads
+the contract states: `getTotalSupply`, `getSymbol`, `getName`,
+`getDecimals` and `balance`. It also has a function `withdraw`, which
+not updates the contract state, but also transfers assets.
+
+After [compiling](/dapps/getting-started#compile-your-contract) the
+contract, Alephium's [Web3 SDK](/dapps/alephium-web3) generates
+typescript class for the contract, which can be loaded in the
```

Comment:
It seems like this sentence is incomplete.

Reply:
nice catch. thanks!